google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -238,29 +254,18 @@ public String getTopicNameString() {
 
     if (batchToSend != null) {
       logger.log(Level.FINER, "Scheduling a batch for immediate sending.");
-      final OutstandingBatch finalBatchToSend = batchToSend;
-      executor.execute(
-          new Runnable() {
-            @Override
-            public void run() {
-              publishOutstandingBatch(finalBatchToSend);
-            }
-          });
+      publishOutstandingBatch(batchToSend);
```

Comment:
Apologies if I've already asked this, but can you please explain the change here and why we're calling publishOutstandingBatch and publishAllOutstanding?

Reply:
The fact that they are in this order is the bug that @kimkyung-goog mentioned. We actually need to reverse the order (and have a bug to do so that he will take care of). The idea is that we'd need to flush all outstanding messages and also flush this message if it is too big to fit within the confines of a single batch.